core/domain/fs_services_test.py

```diff
@@ -182,3 +182,34 @@ def test_compress_image_on_prod_mode_with_small_image_size(self):
                 gae_image_services.get_image_dimensions(
                     micro_image_content),
                 (22, 22))
+
+class FileSystemClassifierDataTests(test_utils.GenericTestBase):
+    """Unit tests for storing, reading and deleting classifier data."""
+    
+    def setUp(self):
+        super(FileSystemClassifierDataTests, self).setUp()
+        self.fs = fs_domain.AbstractFileSystem(
+            fs_domain.GcsFileSystem(
+                feconf.ENTITY_TYPE_EXPLORATION, 'exp_id'))
+        self.classifier_data = {
+            'param1': 40,
+            'param2': [34.2, 54.13, 95.23],
+            'submodel': {
+                'param1': 12
+            }
+        }
+
+    def test_save_and_get_classifier_data(self):
+        """Test that classifier data is stored and retrieved correctly."""
+        fs_services.save_classifier_data(
+            'exp_id', 'job_id', self.classifier_data)
+        classifier_data = fs_services.read_classifier_data('exp_id', 'job_id')
+        self.assertEqual(classifier_data, self.classifier_data)
```

Comment:
Should we add test for the None case line 119 of fs_services?

Reply:
I don't think it is necessary, as returning None means that model is not yet trained and we have other flags to get that information.